### PR TITLE
SGX port tokio 1.34.0

### PR DESCRIPTION
**Pull request by raoulstrackx**
This work is part of the async I/O work. Before upstreaming we should have an internal review.

All credits must go to @mzohreva . He did the heavy lifting, I only rebased things on the latest master.

cc: @BelalH @vn971

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,3 +15,10 @@ members = [
   "tests-build",
   "tests-integration",
 ]
+
+[patch.crates-io]
```
Comment (raoulstrackx) [inline Cargo.toml:19]: Update patches before trying to upstream. Needs [this](https://github.com/fortanix/rust-sgx/pull/515) and [this PR](https://github.com/fortanix/mio/pull/1)

File: ct.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -ex
+repo_root=$(readlink -f $(dirname ${BASH_SOURCE[0]}))
+cd ${repo_root}
```
Comment (raoulstrackx) [inline ct.sh:3]: Part of the "local CI" commit. May be removed before upstreaming

File: tokio/Cargo.toml [hunk 3]
```diff
@@ -164,3 +176,10 @@ rustc-args = ["--cfg", "tokio_unstable", "--cfg", "tokio_taskdump"]
 
 [package.metadata.playground]
 features = ["full", "test-util"]
+
+# set number of threads so tests can run properly
+[package.metadata.fortanix-sgx]
+threads=100
```
Comment (arai-fortanix) [inline tokio/Cargo.toml]: Should there be spaces around the `=` in these parameters?

**Review by arai-fortanix — APPROVED** [review]

**Comment by raoulstrackx** [comment]
Tokio has since been updated to 1.36. Closing PR in favor of #3

**Pull request closed without merging**